### ISSUE TITLE
Gradle run error: Optional(SkipDriveExternal.ProcessResult.ExitStatus.terminated(code: 1))

### ISSUE THREAD

**Issue by lexonerus** [skiptools/skip#24]
**Prerequisites:**
- Mac mini 2018 Intel
- Xcode Version 15.0.1 (15A507)
- Android Studio Giraffe | 2022.3.1 Patch 2
- Apple Swift version 5.9
- java version "17.0.9" 2023-10-17 LTS
- Kotlin version 1.9.10-release-459 (JRE 21.0.1)
- Xcode and AndroidStudio installed in application folder

**Skip Doctor**
```
[✓] Skip version 0.7.16 (= 0.7.16)
[✓] macOS version 14.1 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✓] Xcode version 15.0.1 (> 15.0.0)
[✓] Homebrew version 4.1.18 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✓] Check Skip Updates: 0.7.16
[✓] Skip (0.7.16) checks complete
```

But when I run `skip checkup` Kotlin fails:
```
[✓] Skip version 0.7.16 (= 0.7.16)
[✓] macOS version 14.1 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✓] Xcode version 15.0.1 (> 15.0.0)
[✓] Homebrew version 4.1.18 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✓] Create project hello-skip (5.06s)
[✓] Resolve dependencies (18.22s)
[✓] Build hello-skip (45.23s)
[✓] Test Swift (19.59s)
[✗] Test Kotlin (28.74s)
[✓] Archive iOS ipa (35.74s)
[✓] Assemble HelloSkip-release.ipa (0.03s)
[✓] Verify HelloSkip-release.ipa 23 KB
[✗] Assembling Android apk (4.73s)
[✗] Verify HelloSkip-release.apk: The file “HelloSkip-release-unsigned.apk” couldn’t be opened because there is no such file.
[✗] Skip 0.7.16 checkup (158.55s)
Error: 4 errors
```

Just for test I init new skip xcode project and try to build it, get this error message:
`Gradle run error: Optional(SkipDriveExternal.ProcessResult.ExitStatus.terminated(code: 1))`

What I am doing wrong? Thank for the answer.

**Comment by lexonerus** [comment]
So, what I've done:

1. First of all I found command, to see more logs `skip checkup --verbose` - mostly where fine, but I saw this message: _SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/private/var/folders/f4/l_tb9q055mv4m_8g961c6frc0000gn/T/E6C5E9EF-AD09-4496-8E74-B0C5DC6C4E61/hello-skip/.build/plugins/outputs/hello-skip/HelloSkip/skipstone/local.properties'._
2. It seems that skip can not find AndroidSDK. To fix this issue, I need to set the ANDROID_HOME environment variable to the location of Android SDK. So I opend a terminal and type `echo $ANDROID_HOME`. Result was empty line, so I need to find location of my AndroidSDK and set it to this variable.
3. Then in terminal I typed this `export ANDROID_HOME=/Users/lexone/Library/Developer/AndroidSDK`, in my case AndroidSDK location is in Library/Developer (manually set), you should investigate it by yourself.
4. Now `echo $ANDROID_HOME ` give me this line /Users/lexone/Library/Developer/AndroidSDK, it seems that now all fine.
5. `skip checkup` again
```
[✓] Skip version 0.7.16 (= 0.7.16)
[✓] macOS version 14.1 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✓] Xcode version 15.0.1 (> 15.0.0)
[✓] Homebrew version 4.1.18 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 17.0.9 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✓] Create project hello-skip (1.74s)
[✓] Resolve dependencies (15.63s)
[✓] Build hello-skip (33.52s)
[✓] Test Swift (18.15s)
[✓] Test Kotlin (147.65s)
[✓] Archive iOS ipa (33.21s)
[✓] Assemble HelloSkip-release.ipa (0.02s)
[✓] Verify HelloSkip-release.ipa 23 KB
[✓] Assembling Android apk (138.06s)
[✓] Verify HelloSkip-release.apk 9,5 MB
[✓] Skip 0.7.16 checkup (389.09s)
```
and all set! 
6. Then I tried to rebuild xcode skip project and get same error...

**Comment by lexonerus** [comment]
**SOLVED**.

- In Xcode build phases scripts android sdk path is `~/Library/Android/sdk`, so I moved sdk
- Rebuild project and get another error message, something like "build script phase failed"
- Moved my custom pathes to Derived Data and Archives to default in Xcode
- Init new skip project and all works well

The point was in custom pathes in Android Studio and Xcode. But it was because I have only 128gb on my internal drive. So I has to moved my MacOs to external SSD 512, and it is another story...

![Screenshot 2023-10-31 at 20 20 36](https://github.com/skiptools/skip/assets/26347340/178a6b34-7023-402e-a12b-5e9f189a6fcb)

**Comment by iankoex** [comment]
I have the same issue:
I am currently using skip 0.7.30, Xcode 15.0.1 (15A507) and Android Studio 2022.3.1 Patch 4

- `ANDROID_HOME` is set correctly.
- Xcode custom paths and derived data are set to default.
- All `skip checkup` tests pass
- All `skip doctor` tests pass
- `skip init` initializes the project correctly. 

 The problem arises when building with Xcode, I am getting the error: 
`Gradle run error: Optional(SkipDriveExternal.ProcessResult.ExitStatus.terminated(code: 1))`

**Comment by marcprux** [comment]
What arguments did you pass to `skip init` when you created the project? If you try to build and test the project from the Terminal with `swift test`, do you also get errors?

Are there any other errors reported in the Xcode Issues navigator? If all you see is the "Gradle run error" error, then check the Reports navigator's build failure. Expanding the "Run custom shell script 'Run skip gradle'" should give the full output of the gradle command, which will help us identify the issue.

Here are a could screenshots showing what it might look like:

<img width="1204" alt="Screenshot 2023-11-30 at 08 36 32" src="https://github.com/skiptools/skip/assets/659086/5cd2199d-2af7-4418-bedf-fa82c2ac8c27">


<img width="1204" alt="Screenshot 2023-11-30 at 08 37 56" src="https://github.com/skiptools/skip/assets/659086/637684f6-8c74-4a9d-90bb-7f99702bb87a">

**Comment by TobiasPott** [comment]
For completeness sake:
I've encountered the same error message and couldn't solve it with the ways described (no custom paths anywhere and everything is updated, though Xcode version is newer now). What it solved it for me as the errors hinted towards a missing toolchain for languageVersion 17 in the gradle step, was including the auto toolchain download for gradle.
I've added the following line to the `settings.gradle.kts` file.
```
plugins {
    id("org.gradle.toolchains.foojay-resolver-convention") version("0.7.0")
}
```

This caused the next build to take a while due to the download, but it allowed me to build the project after reopening it in xcode.
